File: backend/src/main/java/com/fitness/fitnessaicoach/repository/UserRepository.java
package com.fitness.fitnessaicoach.repository;

import com.fitness.fitnessaicoach.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID> {

    // Para login en el futuro
    Optional<User> findByEmail(String email);

    // Para validación profesional
    boolean existsByEmail(String email);
}

File: android/fitness-ai-coach-app/app/src/main/java/com/fitness/fitnessaicoach/domain/repository/UserRepository.kt
package com.fitness.fitnessaicoach.domain.repository

import com.fitness.fitnessaicoach.core.result.AppResult
import com.fitness.fitnessaicoach.domain.model.User

interface UserRepository {
    suspend fun getCurrentUser(): AppResult<User>
    suspend fun updateUserProfile(user: User): AppResult<User>
}
